framework/src/modules/interoperability/base_cross_chain_update_command.ts

```diff
@@ -226,7 +214,34 @@ export abstract class BaseCrossChainUpdateCommand<
 		return [ccms, true];
 	}
 
-	protected async afterExecuteCommon(
+	private _verifyRoutingRules(
+		context: CommandExecuteContext<CrossChainUpdateTransactionParams>,
+		isMainchain: boolean,
+		ccm: CCMsg,
+	) {
+		// Sending and receiving chains must differ.
+		if (ccm.receivingChainID.equals(ccm.sendingChainID)) {
+			throw new Error('Sending and receiving chains must differ.');
+		}
+
+		// Processing on the mainchain
+		if (isMainchain) {
+			if (!ccm.sendingChainID.equals(context.params.sendingChainID)) {
```

Comment:
we can preserve comments from LIP
```suggestion
// The CCM must come from the sending chain.
			if (!ccm.sendingChainID.equals(context.params.sendingChainID)) {
```